### ISSUE TITLE
as.party.C50 gives strange result with C5 with weights

### ISSUE THREAD

**Issue by kohleth** [topepo/C5.0#7]
When weights are used in the C5 model, and then the model is converted to a party object, the conversion does not seem to work.

```
library(C50)
library(partykit)

m1=C5.0(Species~.,data=iris,weights=1:nrow(iris))
m1p=as.party(m1)
m1p
predict(m1p,iris[1,])
```
You can see that in this case, the party object is predicting some numerical value instead of class.

**Comment by kohleth** [comment]
Hi,

I don't know if this has to do with the version of C50, but I am using the version from github (0.1.0-25) and my fitted model does not use weights (see fix on issue #6) :

```
> m1=C5.0(Species~.,data=iris,weights=1:nrow(iris))
> summary(m1)
...
Petal.Length <= 4.7:
:...Petal.Length <= 1.9: setosa (16.9)
:   Petal.Length > 1.9: versicolor (45.6/1.4)
Petal.Length > 4.7:
:...Petal.Width > 1.7: virginica (75.8/0.9)
    Petal.Width <= 1.7:
    :...Petal.Length <= 4.9: versicolor (2.7)
        Petal.Length > 4.9: virginica (9/2.1)
```

Of course, this doesn't rule out what you are saying -- that it has to do with how C50 handles weight.

**Comment by topepo** [comment]
Yes, this should be fixed. I'm having some issues with a CRAN release (arcane C issues) but it should be coming soon.

**Comment by kohleth** [comment]
yes, the issue reported by mvculp does not show up in the github version, but the initial issue i reported is still there.

**Comment by mvculp** [comment]
Ok. So, my understanding is that there was an issue with the weights and it has been fixed recently with C5.0, but the fix was not on the latest R (CRAN) site. The recent fix in turn caused a downstream issue with the party conversion.

I have taken the latest version from GitHub and executed it to get the issue reported (specifically the weights become the response in the new version). I updated the as.party.C5.0 to fix what I believe is the issue.
